dspy/experimental/module_graph.py

```diff
@@ -0,0 +1,127 @@
+import dspy
+
+try:
+    import graphviz  # type: ignore
+    graphviz_available = True
+except ImportError:
+    graphviz_available = False
+
+
+class ModuleGraph:
+    def __init__(self, module_name, module):
+        if graphviz_available is False:
+            raise ImportError(
+                """Please install graphviz to use this feature.
+                Run 'pip install graphviz'""")
+
+        self.graph = graphviz.Digraph(format='png')
+        self.nodes = set()
+        self.module_name = module_name
+        self.module = module
+        self.inspect_settings(dspy.settings)
+        self.add_module(self.module_name, self.module)
+
+    def inspect_settings(self, settings):
```

Comment:
curious how setting a LM/RM in using `with dspy.settings.context(...):` ([reference example](https://github.com/stanfordnlp/dspy/blob/abf98a1491ee26a3bfc55c9a8abc340783e55dce/dspy/predict/langchain.py#L74)) of setting LM/RM to impact only the corresponding code block) -  impacts this functionality. is it better to instead have user-configurable parameters when initializing ModuleGraph?

Reply:
and what happens if these are not set?

Reply:
the current implementation only handle default LLMs. Can add LLMs in dspy.context later.

If nothing is set in dspy.settings.configure, then nothing will appear in the graph as expected.